datasets/voc/voc.py

```diff
@@ -0,0 +1,229 @@
+            is_truncated = obj.find("truncated").text == "1"
+            is_difficult = obj.find("difficult").text == "1"
+            bndbox = obj.find("bndbox")
+            xmax = float(bndbox.find("xmax").text)
+            xmin = float(bndbox.find("xmin").text)
+            ymax = float(bndbox.find("ymax").text)
+            ymin = float(bndbox.find("ymin").text)
+            yield {
+                "label": label,
+                "pose": pose,
+                "bbox": [ymin / height, xmin / width, ymax / height, xmax / width],
+                "is_truncated": is_truncated,
+                "is_difficult": is_difficult,
+            }
+
+
+class Voc(datasets.GeneratorBasedBuilder):
+    """Pascal VOC dataset."""
+
+    BUILDER_CONFIGS = [
+        VocConfig(
+            year="2007",
+            description=_VOC_CONFIG_DESCRIPTION.format(year=2007, num_images=9963, num_objects=24640),
+            filenames={
+                "trainval": "VOCtrainval_06-Nov-2007.tar",
+                "test": "VOCtest_06-Nov-2007.tar",
+            },
+            has_test_annotations=True,
+        ),
+        VocConfig(
+            year="2012",
+            description=_VOC_CONFIG_DESCRIPTION.format(year=2012, num_images=11540, num_objects=27450),
+            filenames={
+                "trainval": "VOCtrainval_11-May-2012.tar",
+                "test": "VOC2012test.tar",
+            },
+            has_test_annotations=False,
+        ),
+    ]
+
+    # DEFAULT_CONFIG_NAME = "2012"
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_VOC_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "image": datasets.Image(),
+                    "objects": datasets.Sequence(
+                        {
+                            "label": datasets.ClassLabel(names=_VOC_LABELS),
+                            "bbox": datasets.Sequence(datasets.Value("float32"), length=4),
+                            "pose": datasets.ClassLabel(names=_VOC_POSES),
+                            "is_truncated": datasets.Value("bool"),
+                            "is_difficult": datasets.Value("bool"),
+                        }
+                    ),
+                    "labels": datasets.Sequence(datasets.ClassLabel(names=_VOC_LABELS)),
+                    "labels_no_difficult": datasets.Sequence(datasets.ClassLabel(names=_VOC_LABELS)),
+                }
+            ),
+            homepage=_VOC_URL.format(year=self.config.year),
+            citation=_VOC_CITATION.format(year=self.config.year),
+        )
+
+    def _split_generators(self, dl_manager):
+        paths = dl_manager.download_and_extract(
+            {k: os.path.join(_VOC_DATA_URL, v) for k, v in self.config.filenames.items()}
+        )
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST, gen_kwargs=dict(data_path=paths["test"], set_name="test")
```

Comment:
(nit) we usually define dicts this way
```suggestion
                name=datasets.Split.TEST, gen_kwargs={"data_path": paths["test"], "set_name": "test"}
```